pkg/databricks/client.go

```diff
@@ -343,11 +582,33 @@ func (c *Client) doRequest(ctx context.Context, urlAddress *url.URL, method stri
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusBadRequest {
+			var res struct {
+				Detail  string `json:"detail"`
+				Message string `json:"message"`
+			}
+
+			if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
```

Comment:
It might make sense to check the response content type and only try to decode if it's application/json, otherwise log the response body in the error. The same goes for line 609 below.